docs/advanced_examples/LogisticRegressionTraining.ipynb

```diff
@@ -462,6 +395,339 @@
+       "              random_state=42)</pre></div></div></div></div></div>"
+      ],
+      "text/plain": [
+       "SGDClassifier(fit_encrypted=True, max_iter=15, parameters_range=(-1.0, 1.0),\n",
+       "              random_state=42)"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Initialize the model with parameters\n",
+    "parameters_range = (-1.0, 1.0)\n",
+    "\n",
+    "sgd_clf_binary_fhe = SGDClassifier(\n",
+    "    random_state=RANDOM_STATE,\n",
+    "    max_iter=N_ITERATIONS,\n",
+    "    fit_encrypted=True,\n",
+    "    parameters_range=parameters_range,\n",
+    ")\n",
+    "\n",
+    "# Generate random x and y based on the min and max of X_binary and y_binary\n",
+    "x_min, x_max = X_binary.min(), X_binary.max()\n",
+    "y_min, y_max = y_binary.min(), y_binary.max()\n",
+    "\n",
+    "# Generate random data with the same shape as X_binary and y_binary\n",
+    "x_random = np.random.uniform(x_min, x_max, X_binary.shape)\n",
+    "y_random = np.random.randint(y_min, y_max + 1, y_binary.shape)\n",
+    "\n",
+    "# Fit the model with random data to compile it for production\n",
+    "# This step ensures the model knows the number of features and targets\n",
+    "sgd_clf_binary_fhe.fit(x_random, y_random, fhe=\"disable\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from concrete.ml.deployment import FHEModelDev\n",
+    "\n",
+    "# Save the training FHE circuit for production\n",
+    "fhe_dev = FHEModelDev(\"fhe_training_sgd\", sgd_clf_binary_fhe)\n",
+    "fhe_dev.save(mode=\"training\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from concrete.ml.deployment import FHEModelClient\n",
+    "\n",
+    "# On the client side, load the circuit.zip with the information to create\n",
+    "# - the key\n",
+    "# - the pre and post processing functions\n",
+    "\n",
+    "fhe_client = FHEModelClient(\"fhe_training_sgd\")\n",
+    "fhe_client.load()\n",
+    "serialized_evaluation_key = fhe_client.get_serialized_evaluation_keys()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from concrete.ml.deployment import FHEModelServer\n",
+    "\n",
+    "# On the server side, we load the server.zip which contain the training model\n",
+    "fhe_server = FHEModelServer(\"fhe_training_sgd\")\n",
+    "fhe_server.load()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define utils function to evaluate the model, print it's accuracy\n",
+    "\n",
+    "\n",
+    "def model_inference(weights, bias, X):\n",
```

Comment:
see comment https://github.com/zama-ai/concrete-ml/pull/665/files#r1618751835, I don't think we should need to re-implement the inference like this

Reply:
Yes I opened https://github.com/zama-ai/concrete-ml-internal/issues/4464.